pageserver/src/bin/pageserver.rs

```diff
@@ -76,8 +77,23 @@ fn main() -> Result<()> {
                 .help("Additional configuration overrides of the ones from the toml config file (or new ones to add there).
                 Any option has to be a valid toml document, example: `-c=\"foo='hey'\"` `-c=\"foo={value=1}\"`"),
         )
+        .arg(
+            Arg::new("enabled-features")
+                .long("enabled-features")
+                .takes_value(false)
+                .help("Show enabled compile time features"),
+        )
         .get_matches();
 
+    if arg_matches.is_present("enabled-features") {
+        let features: &[&str] = &[
+            #[cfg(feature = "failpoints")]
+            "failpoints",
+        ];
+        println!("available features: {:?}", features);
+        return Ok(());
+    }
+
```

Comment:
Neat! Although I'd print this info in machine-friendly format e.g. json. Something along the lines:

```json
{ "features": ["failpoints"] }
```

That would be trivially parsable from python tests (`import json; json.loads("...")`).